app/src/main/java/com/android/sample/ui/camera/Gallery.kt

```diff
@@ -53,24 +73,86 @@ fun ProfileImage(userId: String, modifier: Modifier = Modifier, imageViewModel:
         })
   }
 
+  LaunchedEffect(bitmap) {
+    if (bitmap != null && editing) {
+      isImageRemoved = false
+    } else if (bitmap == null && imageUrl == null && editing) {
+      isImageRemoved = true
+    }
+  }
+  LaunchedEffect(imageUrl) {
+    if (imageUrl != null && editing) {
+      isImageRemoved = false
+    }
+  }
+
+  val defaultImageId = remember(userId) { randomDefaultProfileImage() }
+  // Determine which painter to use
   val painter =
-      rememberAsyncImagePainter(
-          model =
-              ImageRequest.Builder(context)
-                  .apply {
-                    // Only parse the URI if the URL is not null and not empty
-                    if (!imageUrl.isNullOrEmpty()) {
-                      data(Uri.parse(imageUrl))
-                    } else {
-                      data(null) // Handle null or empty string by not attempting to load anything
-                    }
-                    crossfade(true)
-                  }
-                  .build())
+      when {
+        editing && bitmap != null -> {
+          // Use the bitmap provided when editing
+          rememberAsyncImagePainter(model = ImageRequest.Builder(context).data(bitmap).build())
+        }
+        editing && isImageRemoved -> {
+          // Use the default monkey picture when the image is marked as removed
+          painterResource(id = defaultImageId)
+        }
+        !imageUrl.isNullOrEmpty() -> {
+          // Use the profile image URL from Firebase Storage, with caching via Coil
+          rememberAsyncImagePainter(
+              model =
+                  ImageRequest.Builder(context).data(Uri.parse(imageUrl)).crossfade(true).build())
+        }
+        else -> {
+          // Fallback to the default image
+          painterResource(id = defaultImageId)
+        }
+      }
+
+  Box() {
+    Image(
+        painter = painter,
+        contentDescription = "Profile Image",
+        modifier = modifier,
+        contentScale = ContentScale.Crop)
+
+    // Show the "X" button to remove the image when editing
+    if (editing && !isImageRemoved) {
+      IconButton(
+          onClick = {
+            isImageRemoved = true // Mark the image as removed
+            onRemoveImage() // Trigger external action if needed
+          },
+          modifier =
+              Modifier.align(Alignment.TopCenter)
+                  .size(24.dp)
+                  .background(color = Color.Red, shape = CircleShape)) {
+            Icon(
+                imageVector = Icons.Default.Close,
+                contentDescription = "Remove Image",
+                tint = Color.White)
+          }
+    }
+  }
+}
+
```

Comment:
Introduced a new function randomDefaultProfileImage to randomly select a default profile image.